packages/components/tests/dummy/app/controllers/components.js

```diff
@@ -0,0 +1,24 @@
+import Controller from '@ember/controller';
+import { scheduleOnce } from '@ember/runloop';
+
+function replaceDummyStates() {
+  document.querySelectorAll('[mock-state-value]').forEach(function (element) {
+    let target;
+    if (element.attributes['mock-state-selector']) {
+      target = element.querySelector(
+        element.attributes['mock-state-selector'].value
+      );
+    } else {
+      target = element;
+    }
+    const state = element.attributes['mock-state-value'].value;
+    target.classList.add(`mock-${state}`);
+  });
```

Comment:
The logic here is this: we have two possible custom HTML attributes (not prefixed with `data` intentionally, even if it's not valid HTML for our purpose is OK)
- `mock-state-value` is the value of the _state_ that we want to use, to apply a class `mock-[state]` to an element, so that a custom CSS class can be used to emulate a particular state in a component (typically pseudo states like `:hover` or `:focus`) in a showcase
- `mock-state-selector` is the DOM selectors we want to use (relative to the element to which is applied) to select the _target_ element on which dynamically apply the mock class at runtime; if no selector is defined, we simply use the element with the `mock-state-value` attribute itself